library/psa_crypto.c

```diff
@@ -823,12 +846,17 @@ static psa_status_t psa_export_ecp_key( psa_key_type_t type,
                 return status;
         }
 
-        return( mbedtls_to_psa_error(
+        status = mbedtls_to_psa_error(
                     mbedtls_ecp_point_write_binary( &ecp->grp, &ecp->Q,
                                                     MBEDTLS_ECP_PF_UNCOMPRESSED,
                                                     data_length,
                                                     data,
-                                                    data_size ) ) );
+                                                    data_size ) );
+
+        if( status != PSA_SUCCESS )
```

Comment:
Blocker: `mbedtls_ecp_write_key` fails with partially written data if the buffer is too large. Indeed the output buffer needs to be wiped on failure. It's done for RSA in `psa_export_rsa_key` and it's done for public EC keys so only the private EC case is missing.

Thanks, I'd missed that one in my review.

Reply:
I'll add it, but I'll also say that I don't really get where it would partially fail. `ecp_write_key` only contains a single call to `mpi_write_binary`, which in turn returns early if the write would fail, before actually attempting a write. When the writing has begun, it'll always return success.

Reply:
Oh, right. I'd read `mpi_write_binary` too quickly. So it isn't a blocker (nothing immediately broken), but it's better to wipe the buffer because the atomic behavior of `mpi_write_binary` is not guaranteed by the API.